netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -157,12 +157,11 @@ public static NettyChannelBuilder forTarget(String target) {
    */
   @CheckReturnValue
   public static NettyChannelBuilder forTarget(String target, ChannelCredentials creds) {
-    ProtocolNegotiators.FromChannelCredentialsResult result = ProtocolNegotiators.from(creds);
+    FromChannelCredentialsResult result = ProtocolNegotiators.from(creds);
     if (result.error != null) {
       throw new IllegalArgumentException(result.error);
     }
-    return new NettyChannelBuilder(
-        target, result.negotiator, result.callCredentials);
+    return new NettyChannelBuilder(target, creds);
```

Comment:
No. This is not appropriate. Don't process the creds twice. Either have the constructor throw or pass `result` to the constructor. Ditto for the provider. Ditto for OkHttp.

Reply:
Refactored the args.